### ISSUE TITLE
Crash on deleting BB-patterns while previewing.

### ISSUE THREAD

**Issue by Spekular** [LMMS/lmms#1683]
**Edit Jan 27 @tresf:  reproduced here:**  [`#1683 (comment)`](https://github.com/LMMS/lmms/issues/1683#issuecomment-71667995)

---

I can't seem to reproduce anymore but maybe someone else can? Deleting a BB-track as the BB-pattern is previewed in the BB-editor causes a crash. The steps I originally followed:
Create a BB-pattern from BB-editor.
Populate pattern with hits/notes
Preview pattern in BB-editor
Delete respective BB-track in song editor.

Some things that may have had to do with it:
All the elements of the pattern were samples
I had 7 samples, I was using 4 in that particular pattern
There was at least one hit/note every other tombstone

I know this is horribly vague and extremely hard to fix if I can't even reproduce it. Perhaps if anyone else experiences a crash like this they could write down their steps and that way we could narrow it down.

**Comment by Sti2nd** [comment]
> Preview pattern in BB-editor

Like in view with the eyes or play it? Could not reproduce when playing and viewing.

**Comment by Spekular** [comment]
@sti2nd as in pressing the play button inside the bb-editor.

**Comment by tresf** [comment]
Yeah, without steps to reproduce, this will eventually be closed and left with the archives of time. :)

**Comment by tresf** [comment]
P.S. Tried reproducing on 1.1.0 to no avail. :+1:

**Comment by badosu** [comment]
@Spekular I can't find but I know what you're talking about.

I experience this crash a lot, I can't seem to find the other issue about this though.

Add two kickers and hit play and tweak with their settings, happens a lot to me. It's a really frustrating bug.

**Comment by Spekular** [comment]
@badosu I have steps to reproduce! Worked 3/3 tries so far. Typing them up.

**Comment by badosu** [comment]
Thanks for the detailed steps @Spekular, greatly aprecciated.

Gonna take a look at this when possible.

**Comment by Sti2nd** [comment]
> Right click watsyn's volume in BB-editor, connect to controller > user controller 1

A bit off topic perhaps: What's up with the controllers in LMMS 1.1.0? They lag like the land of the dead.

**Comment by tresf** [comment]
> A bit off topic perhaps: What's up with the controllers in LMMS 1.1.0? They lag like the land of the dead.

Yes, off topic (and no idea).

**Comment by badosu** [comment]
@Spekular Is'nt there an easier way with less steps to reproduce the crash? It's being terrible to debug using your description.

**Comment by Spekular** [comment]
There might be, but I only tested this specific way three times before I
had to do other things. I'm certain it can be generalized further.

**Comment by Spekular** [comment]
After more testing, here are generalized instructions:
Open a new project
Delete everything except B/B 1
Delete Kicker in B/B 1
Add sample to B/B 1, with hits on beat 1 and 3
Add a peak controller to the sample, enable mute effect
Add a new B/B pattern from B/B editor
Add watsyn phase bass to B/B 2, add a one bar note
Connect watsyn volume to peak controller
Add one bar of B/B 1 & 2
Play song
Delete B/B 2

**Comment by tresf** [comment]
@Spekular thanks for simplification but I couldn't reproduce and your instructions are hard to follow.

 * You should use `1., 2., etc` numbers so people remember where they left off.
 * You refer to BB1 but you never deleted BB0.
 * You say "Connect watsyn to peak controller" but you really connect it to Controller 1, which isn't obvious for people who don't use controllers that often.

**Comment by Spekular** [comment]
@tresf I ought to have mentioned that this is stable 1.1 and I haven't
tested master yet. I will try to do so. If it exists there I'll see if I
can provide a project file that can be used to reproduce.

**Comment by tresf** [comment]
> @tresf I ought to have mentioned that this is stable 1.1 and I haven't
tested master yet. I will try to do so. If it exists there I'll see if I
can provide a project file that can be used to reproduce.

Yes, I assumed so. I was testing on stable-1.1 as well.

**Comment by Spekular** [comment]
@tresf there's something very strange happening here that I thought I was imagining. I can save right before playing and deleting and get a crash, but loading the saved project and playing and deleting again doesn't cause a crash.
Once again better instructions:
1. Open a project, delete everything except Beat/Bassline 0
2. Add a four on the floor pattern to kicker, and add a peak controller. Enable "mute effect"
3. Create a new Beat/Bassline pattern, and add any instrument to it
4. Add a four measure note to the instrument in Beat/Bassline 1, and connect it's volume to controller 1 (the peak controller)
5. Add four measures of both Beat/Basslines.
6. Play the song, delete Beat/Bassline 1

**Comment by Sti2nd** [comment]
Reproduced on Win8 64 bit.

I noticed again that the connected knob start to lag when connected to a controller. Could be related.
